### PR TITLE
Alpha/buttons-update

### PR DESCRIPTION
**Pull request by iondrimba**
**CHANGELOG** :memo:

* Updating button component style
* Updating button documentation

captura de tela 2017-04-20 as 2 36 19 pm

### CODE REVIEW

File: docs/demo/gaiden-css/components/buttons/sizes/demo.html [hunk 1]
```diff
@@ -4,17 +4,18 @@
   <meta charset="utf-8">
   <link rel="stylesheet" href="/gaiden-css/gaiden.css">
   <link rel="stylesheet" href="/gaiden-css/base.css">
+  <title>Buttons</title>
 </head>
 
 <body>
 <div>
-  <button class="btn btn--high-contrast btn--mini">MINI</button>
+  <button class="btn btn--high-contrast btn--mini">MINI</button><br><br>
```
Comment (eduardojmatos) [inline docs/demo/gaiden-css/components/buttons/sizes/demo.html]: 👁 

hahaha it's ok to use `<br>` here, just a joke 😄

File: src/scss/components/_button.scss [hunk 2]
```diff
@@ -16,69 +16,63 @@
 **/
 
 /**
-  * @styles buttons_mc Medium Contrast
```
Comment (eduardojmatos) [inline src/scss/components/_button.scss]: If we'll remove this documentation, let's remove the button too. 

~Break changes is coming~

File: src/scss/generic/_body.scss [hunk 3]
```diff
@@ -1,9 +1,12 @@
 body {
+  -webkit-font-smoothing: antialiased;
```
Comment (eduardojmatos) [inline src/scss/generic/_body.scss]: Nice! Let's add `-moz-osx-font-smoothing: grayscale` too 😄

File: src/scss/settings/_buttons.scss [hunk 4]
```diff
@@ -81,19 +86,57 @@ $buttonTypes: (
...
+      )
     )
   ),
   high-contrast-reversed: (
     default: (
-      background: (color: air, opacity: .9),
-      textColor: punch
+      background: (color: air),
+      textColor: punch,
+      shadow: (
```
Comment (eduardojmatos) [inline src/scss/settings/_buttons.scss]: This can be inline (`shadow: ( kind: normal ) )` and in another occurrences of this case (L124, L137, etc.)

File: src/scss/tools/mixins/_generate_button_types.scss [hunk 5]
```diff
@@ -23,15 +37,27 @@
     }
 
     @if map-has-key($buttonTypeConfig, border) {
-      border: solid 1px get-color(map-get($buttonTypeConfig, border));
+      $config: map-get($buttonTypeConfig, border);
+      $style: map-get($config, style);
```
Comment (eduardojmatos) [inline src/scss/tools/mixins/_generate_button_types.scss]: You can use `map-deep-get($buttonTypeConfig, border, style)` here.

File: src/scss/tools/mixins/_generate_button_types.scss [hunk 6]
```diff
@@ -23,15 +37,27 @@
...
+      $color: map-get($config, color);
```
Comment (eduardojmatos) [inline src/scss/tools/mixins/_generate_button_types.scss]: The same above 😄

File: src/scss/components/_button.scss (hunk 2 above)
Comment (iondrimba) [inline src/scss/components/_button.scss]: @eduardojmatos Fair enough

File: src/scss/tools/mixins/_generate_button_types.scss (hunk 5 above)
Comment (iondrimba) [inline src/scss/tools/mixins/_generate_button_types.scss]: @eduardojmatos Great tip, will change it

File: src/scss/components/_button.scss [hunk 7]
```diff
@@ -133,6 +130,14 @@
     }
   }
 
+  &--full-card {
+    border-top: get-border(default);
+    border-bottom-right-radius: $borderRadius;
+    border-bottom-left-radius: $borderRadius;
```
Comment (MarcoBrunoBR) [inline src/scss/components/_button.scss]: Now we have two border radius. What do you think of making clear what the $borderRadius does? Maybe you changed his name?

Comment (iondrimba) [inline src/scss/components/_button.scss]: @MarcoBrunoBR I can't really change the name of $borderRadius on this PR because its the default variable for all the existing borders,  also, this change will affect files not related to this PR. We can discuss about the naming convention after the merge to v11-alpha.  Do you agree? @eduardojmatos @wbotelhos

Comment (MarcoBrunoBR) [inline src/scss/components/_button.scss]: Fair enough @iondrimba

**Comment by iondrimba** [comment]
@MarcoBrunoBR @eduardojmatos Anything else needs to change?

File: src/scss/generic/_body.scss (hunk 3 above)
Comment (eduardojmatos) [inline src/scss/generic/_body.scss]: @iondrimba ☝️

File: src/scss/components/_button.scss (hunk 7 above)
Comment (eduardojmatos) [inline src/scss/components/_button.scss]: Yes, we can do this on another PR :smile:

File: src/scss/generic/_body.scss (hunk 3 above)
Comment (iondrimba) [inline src/scss/generic/_body.scss]: @eduardojmatos it's already there, I don't know why is not showing the change. Take a look a the file. 👍

File: src/scss/generic/_body.scss [hunk 8]
```diff
@@ -1,9 +1,13 @@
...
+  -moz-osx-font-smoothing: grayscale;
```
Comment (iondrimba) [inline src/scss/generic/_body.scss]: @eduardojmatos Here 👍
